src/errorpage.cc

```diff
@@ -828,8 +828,8 @@ ErrorState::Dump(MemBuf * mb)
     if (auth_user_request.getRaw() && auth_user_request->denyMessage())
         str.appendf("Auth ErrMsg: %s\r\n", auth_user_request->denyMessage());
 #endif
-    if (dnsError.size() > 0)
-        str.appendf("DNS ErrMsg: %s\r\n", dnsError.termedBuf());
+    if (dnsError && dnsError->length() > 0)
+        str.appendf("DNS ErrMsg: %s\r\n", dnsError->c_str());
```

Comment:
Please avoid ```c_str()``` whenever possible.
```suggestion
        str.appendf("DNS ErrMsg: " SQUIDSBUFPH "\r\n", SQUIDSBUFPRINT(*dnsError));
```